### ISSUE TITLE
Constructor on type 'Xunit.Sdk.InlineDataDiscoverer' not found.

### ISSUE THREAD

**Issue by P-A-R-U-S** [xunit/xunit#819]
I'm geting following error when I run test in Debug mode:

Constructor on type 'Xunit.Sdk.InlineDataDiscoverer' not found.

at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes, StackCrawlMark& stackMark)

**Comment by bradwilson** [comment]
Is it fatal?

**Comment by onovotny** [comment]
Just a guess, but I think people are hitting this because we ship symbols in the nuget. This causes the debugger to think that it's "my code" and bubbles up the exceptions that would otherwise be ignored when "just my code" is enabled.

We may want to look at whatever mechanisms CoreFX/Roslyn, etc will use to provide their symbols. I don't believe they were leaning towards including them in the main nuget just for that reason.

**Comment by strake7** [comment]
To contribute to this finding  this exception bubbled for me as well when using "dotnet-test-xunit": "2.2.0-preview2-build1029",  "XUnit": "2.2.0-rc4-build3536". The exception is not fatal and enabling "just my code"  will thus ignore the exception.

**Comment by zxxc** [comment]
enabling "just my code" fixing this issue for me too

**Comment by bradwilson** [comment]
Closed for age.

**Comment by Mousavi310** [comment]
Thanks to @strake7 For JetBrains Rider, just disabling `CLR Exception Breakpoints` solve my problem.

**Comment by EgilSandfeld** [comment]
For me, enabling **Only break on exceptions thrown from user code** in `Breakpoints` settings -> `CLR Exception Breakpoints` fixed this issue
https://i.imgur.com/AyRyIqY.png

**Comment by workabyte** [comment]
I just ran into this issue using JetBrains Rider and found that deleting `/.idea/workspace.xml` file fixed these errors. 

I will update once I find the root content in the file and maybe this solves for good

**Comment by Eilon** [comment]
A few years later, chiming in here. These first-exceptions have been bugging me whenever I debug tests for a project I work on. I've 'fixed' this by having copies of the original xUnit types, but I've also added the constructor that takes in the xUnit Message Sink, so you don't get the reflection exception.

Here's a PR where I fixed it for the project I work on: https://github.com/dotnet/maui/pull/24130

Hopefully someone finds this useful!
